src/internal/connector/onedrive/permission.go

```diff
@@ -210,7 +210,14 @@ func restorePermissions(
 		pbody.SetRequireSignIn(&rs)
 
 		rec := models.NewDriveRecipient()
-		rec.SetEmail(&p.Email)
+		if p.EntityID != "" {
+			rec.SetObjectId(&p.EntityID)
+		} else {
+			// Previous versions used to only store email for a
+			// permissions. Use that if id is not found.
+			rec.SetEmail(&p.Email)
+		}
```

Comment:
Should just not bother with this as permissions is still in experimental?

Reply:
it seems like it would be easier to guarantee a backup would have one or the other. Is it possible to share items with someone that won't have an ID?

@vkamra do we have a gmail account or something we can attempt sharing with to see if we'll get an ID in the returned permission?

Reply:
> it seems like it would be easier to guarantee a backup would have one or the other. Is it possible to share items with someone that won't have an ID?

Ahh, you are right. It totally can have permissions for any random email outside of the system which will not have id. Posting a sample response here.

``` json
{
    "id": "c68a61e2-73fd-4f7d-85c9-fa4d920d6de6",
    "roles": [
        "write"
    ],
    "hasPassword": false,
    "grantedToIdentitiesV2": [
        {
            "user": {
                "@odata.type": "#microsoft.graph.sharePointIdentity",
                "displayName": "email@provider.com",
                "email": "email@provider.com"
            }
        }
    ],
    "grantedToIdentities": [
        {
            "user": {
                "displayName": "email@provider.com",
                "email": "email@provider.com"
            }
        }
    ],
    "link": {
        "scope": "users",
        "type": "edit",
        "webUrl": "https://1...2-my.sharepoint.com/:u:/g/personal/joh...ft_com/Ef...nw",
        "preventsDownload": false
    }
}
```

Reply:
It sends an email to you when the file is shared, clicking that open a web ui where it asks you for email. When you enter the email, you get a code sent to that account which you can use to open it.

![2023-03-01-15-44-06](https://user-images.githubusercontent.com/14259816/222110118-ddb2fb8f-b79b-424a-9236-3d9b1afe31d7.png)

Reply:
Closing this as this is specific to link shared which we are not handling as of now.

Reply:
agree with punting all of the details on this until if/when we handle link shares. From the code though, it seems like it may workout alright anyway though? If the ID isn't populated then we'll pull the email and send that when we restore permissions which should (in theory) trigger the whole workflow you described above?

Reply:
Yup, that will work. The only difference is that link shares are in `grantedToIdentitiesV2` instead of `grantedToV2` which is what we are currently backing up.

Just for context, the primary reason that I did not originally work on link shares was because we could not create the same links as before and that becomes pointless when the link was shared externally. Even with link shares to emails, it would end up emailing the person about the new link(for every item with a custom permission).